test/tests/test_sql_fields.py

```diff
@@ -575,3 +577,23 @@ def test_retrieves_eui_type(self):
         instance = MACArrayTestModel.objects.get(id=instance.id)
         self.assertEqual(instance.field, [EUI('00:aa:2b:c3:dd:44')])
         self.assertIsInstance(instance.field[0], EUI)
+
+
+class TestAggegate(TestCase):
+    @skipIf(VERSION < (1, 9), 'Postgres aggregates not supported in Django < 1.9')
+    def test_aggregate(self):
+        from django.contrib.postgres.aggregates import ArrayAgg
+
+        network = IPv4Network('10.10.10.10/32')
+        inet = IPv4Interface('10.20.30.20/32')
+
+        parent = AggregateTestModel.objects.create()
+        network_qs = AggregateTestModel.objects.annotate(agg_network=ArrayAgg('children__network'))
+        inet_qs = AggregateTestModel.objects.annotate(agg_inet=ArrayAgg('children__inet'))
+
+        self.assertEqual(network_qs[0].agg_network, [])
+        self.assertEqual(inet_qs[0].agg_inet, [])
+
+        AggregateTestChildModel.objects.create(parent=parent, network=network, inet=inet)
+        self.assertEqual(network_qs[0].agg_network, [network])
+        self.assertEqual(inet_qs[0].agg_inet, [inet])
```

Comment:
Ideally, tests should test one thing at a time so that there is more precision in a failure and the tests are more concise.